code/game/mecha/mech_bay.dm

```diff
@@ -105,97 +46,55 @@
 	component_parts += new /obj/item/weapon/stock_parts/capacitor/super(null)
 	component_parts += new /obj/item/stack/cable_coil(null, 1)
 	RefreshParts()
+	recharging_turf = get_step(loc, dir)
 
 /obj/machinery/mech_bay_recharge_port/RefreshParts()
 	var/MC
 	for(var/obj/item/weapon/stock_parts/capacitor/C in component_parts)
 		MC += C.rating
-	if(pr_recharger)
-		pr_recharger.max_charge = MC * 10
-	if(recharge_console)
-		recharge_console.voltage = MC * 10
-
-/obj/machinery/mech_bay_recharge_port/Destroy()
-	qdel(pr_recharger)
-	pr_recharger = null
-	recharge_console.recharge_port = null
-	recharge_floor.recharge_port = null
-	return ..()
+	max_charge = MC * 25
 
 /obj/machinery/mech_bay_recharge_port/attackby(obj/item/I, mob/user, params)
 	if(default_deconstruction_screwdriver(user, "recharge_port-o", "recharge_port", I))
 		return
 
 	if(default_change_direction_wrench(user, I))
+		recharging_turf = get_step(loc, dir)
 		return
 
 	if(exchange_parts(user, I))
 		return
 
-	default_deconstruction_crowbar(I)
-
-/obj/machinery/mech_bay_recharge_port/proc/start_charge(var/obj/mecha/recharging_mecha)
-	if(stat&(NOPOWER|BROKEN))
-		recharging_mecha.occupant_message("<font color='red'>Power port not responding. Terminating.</font>")
-		return 0
-	else
-		if(recharging_mecha.cell)
-			recharging_mecha.occupant_message("Now charging...")
-			pr_recharger.start(list(src,recharging_mecha))
-			return 1
-		else
-			return 0
-
-/obj/machinery/mech_bay_recharge_port/proc/stop_charge()
-	if(recharge_console && !recharge_console.stat)
-		update_icon()
-	pr_recharger.stop()
-	return
-
-/obj/machinery/mech_bay_recharge_port/proc/active()
-	if(pr_recharger.active())
-		return 1
-	else
-		return 0
-
-
-/obj/machinery/mech_bay_recharge_port/power_change()
-	if(powered())
-		stat &= ~NOPOWER
-	else
-		spawn(rand(0, 15))
-			stat |= NOPOWER
-			pr_recharger.stop()
-	return
-
-/obj/machinery/mech_bay_recharge_port/proc/set_voltage(new_voltage)
-	if(new_voltage && isnum(new_voltage))
-		pr_recharger.max_charge = new_voltage
-		return 1
-	else
-		return 0
+	if(default_deconstruction_crowbar(I))
+		recharge_console.recharge_port = null
+		recharge_console.update_icon()

```

Comment:
The two above are unecessary. When you deconstruct the machine, it completely deletes/destroys it--these procs are called in `Destroy`